MSAL/src/public/MSALError.h

```diff
@@ -117,235 +115,223 @@ extern NSString *MSALBrokerVersionKey;
+    MSALErrorInvalidScope                       = -42005,
+    MSALErrorUnauthorizedClient                 = -42006,
     
     /*!
      The server returned an unexpected http response. For instance, this code
      is returned for 5xx server response when something has gone wrong on the server but the
      server could not be more specific on what the exact problem is.
      */
-    MSALErrorUnhandledResponse              = -42007,
-
-    MSALErrorServerDeclinedScopes           = -42008,
-
-    /*! 
-        The passed in authority URL does not pass validation.
-        If you're trying to use B2C, you must disable authority validation by
-        setting validateAuthority of MSALPublicClientApplication to NO.
-     */
-    MSALErrorFailedAuthorityValidation = -42010,
+    MSALErrorUnhandledResponse                  = -42007,
     
     /*!
-        Interaction required errors occur because of a wide variety of errors
-        returned by the authentication service. In all cases the proper response
-        is to use a MSAL interactive AcquireToken call with the same parameters.
-        For more details check MSALOAuthErrorKey and MSALOAuthErrorDescriptionKey
-        in the userInfo dictionary.
+     The user or application failed to authenticate in the interactive flow.
+     Inspect MSALOAuthErrorKey and MSALErrorDescriptionKey in the userInfo
+     dictionary for more detailed information about the specific error.
      */
-    MSALErrorInteractionRequired        = -42100,
-    MSALErrorMismatchedUser             = -42101,
-    MSALErrorNoAuthorizationResponse    = -42102,
-    MSALErrorBadAuthorizationResponse   = -42103,
+    MSALErrorAuthorizationFailed                = -42104,
     
     /*!
-        The user or application failed to authenticate in the interactive flow.
-        Inspect MSALOAuthErrorKey and MSALErrorDescriptionKey in the userInfo
-        dictionary for more detailed information about the specific error.
+     Response was received in a network call, but the response body was invalid.
+     
+     e.g. Response was to be expected a key-value pair with "key1" and
+     the json response does not contain "key1" elements
+     
      */
-    MSALErrorAuthorizationFailed = -42104,
+    MSALErrorInvalidResponse                    = -42600,
     
     /*!
-        MSAL received a bad token response, it didn't contain an access token or id token.
-        Check to make sure your application is consented to get all of the scopes you are asking for.
+     The requested resource is protected by an Intune Conditional Access policy.
+     The calling app should integrate the Intune SDK and call the remediateComplianceForIdentity:silent: API,
+     please see https://aka.ms/intuneMAMSDK for more information.
      */
-    MSALErrorBadTokenResponse = -42105,
+    MSALErrorServerProtectionPoliciesRequired   = -42603,
+    
+    // Workplace join is required to proceed
+    MSALErrorWorkplaceJoinRequired              =  -42710,
```

Comment:
is this one ever returned to MSAL?

Reply:
This one produced in the local interactive controller. I think we can get this one in MSAL, or I am wrong?